src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,93 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+  const {
+    id, title, completed, userId,
+  } = todo;
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const selectedUser = await getUser(userId);
+
+        setUser(selectedUser);
+      } catch (error) {
+        throw new Error('Error while selecting user');
```

Comment:
it would be better to set error somewhere and show the message. now it will just break the whole app